components/automate-cli/pkg/verifyserver/services/startmockserverservice/startmockserverservice.go

```diff
@@ -0,0 +1,182 @@
+package startmockserverservice
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/constants"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+)
+
+// MockServerService provides functionality to start mock servers.
+type MockServerService struct {
+	MockServers []*models.Server
+}
+
+type IStartMockServersService interface {
+	StartMockServer(cfg *models.StartMockServerRequestBody) error
+	GetMockServers() []*models.Server
+}
+
+func New() IStartMockServersService {
+	return &MockServerService{}
+}
+
+// func to get mock server list
+func (s *MockServerService) GetMockServers() []*models.Server {
+	return s.MockServers
+}
+
+// StartMockServer starts a mock server of the given type and port.
+func (servers *MockServerService) StartMockServer(cfg *models.StartMockServerRequestBody) error {
+	var myServer *models.Server
+	var err error
+	switch cfg.Protocol {
+	case constants.TCP:
+		myServer, err = servers.StartTCPServer(cfg.Port)
+	case constants.UDP:
+		myServer, err = servers.StartUDPServer(cfg.Port)
+	case constants.HTTPS:
+		myServer, err = servers.StartHTTPSServer(cfg.Port, cfg.Cert, cfg.Key)
+	default:
+		err = errors.New("unsupported protocol")
+	}
+	if err != nil {
+		return err
+	}
+	servers.MockServers = append(servers.MockServers, myServer)
+	return nil
+}
+
+func (s *MockServerService) StartTCPServer(port int) (*models.Server, error) {
+	// create a TCP listener on the specified port and
+	// save the listener instance in the handler struct
+	listener, err := net.Listen(constants.TCP, fmt.Sprintf("localhost:%d", port))
+
+	if err != nil {
+		return nil, errors.New(err.Error())
+	}
+
+	log.Printf("TCP server started on port %d", port)
+
+	go func() {
+		for {
+			conn, err := listener.Accept()
+			if err != nil {
+				log.Println("Error accepting connection:", err)
+				return
+			}
+			go s.HandleTCPRequest(conn)
+			conn.Close()
+		}
+	}()
+
+	return &models.Server{
+		Port:        port,
+		ListenerTCP: listener,
+		ListenerUDP: nil,
+		Protocol:    constants.TCP,
+	}, nil
+}
+
+func (s *MockServerService) HandleTCPRequest(conn net.Conn) {
+	defer conn.Close()
+	buf := make([]byte, 1024)
+	_, err := conn.Read(buf)
+	if err != nil {
+		fmt.Println("Error reading data from connection:", err)
+		return
+	}
+
+	// time := time.Now().Format(time.ANSIC)
+	responseStr := fmt.Sprintf("Your message is: %v.", string(buf[:]))
+	conn.Write([]byte(responseStr))
+
+	// close conn
+	conn.Close()
```

Comment:
this connection is getting close three times here and then at line 87 and line 74

Reply:
I am retaining the one in side handleUDP and removing others